adding nodegroup documentation (#2187)
kubeflow/website: content/en/docs/aws/pipeline.md

>            image: gcr.io/ml-pipeline/frontend:0.2.0

Reviewer: Line 3:

- Refactor to bring it in line with the title of the guide

```diff
- description = "Customize Kubeflow Pipelines to use AWS Services"
+ description = "Configure Kubeflow Pipelines to use on AWS Services"
```

Lines 9-10:

- Refactor
- "v1.1.0" -> "Kubeflow v1.1"
- Typo in "communitation" -> "communication", "walkaround" -> "workaround"

```diff
- In v1.1.0, in-cluster communitation from notebook to Kubeflow Pipeline is not supported in this phase. In order to use `kfp` as previous, user needs to pass a cookie to KFP for communication as a walkaround.
You can follow following steps to get cookie from your browser after you login Kubeflow. Following examples uses Chrome browser.
+ In Kubeflow v1.1, there is currently no support for in-cluster communitation from a Jupyter notebook to Kubeflow Pipelines. In order to use `kfp`, you need to pass a cookie to Kubeflow Pipelines for communication as a workaround.
Use the following steps to get a cookie from your browser after you log into Kubeflow. These examples use the Chrome browser.
```

Lines 23-25:
- Refactor:

```diff
- Once you get cookie, you can easily authenticate `kfp` by passing the `cookies`. Please look at code snippets based on the manifest you use.
- To get `<aws_alb_host>`, please run `kubectl get ingress -n istio-system` and get value from `ADDRESS` field.		
+ Once you get a cookie, you can easily authenticate `kfp` by passing the `cookies`. Check the code snippets below based on the manifest you are using.
+ Run the command `kubectl get ingress -n istio-system` to get the value from the `ADDRESS` field and replace `<aws_alb_host>` with it.
```

Line 49:

```diff
- Please look at this [PR](https://github.com/kubeflow/kfctl/issues/140#issuecomment-578837304) to do programmatic authentication with Dex.
+ To do programmatic authentication with Dex, refer to this [PR](https://github.com/kubeflow/kfctl/issues/140#issuecomment-578837304).
```

Lines 54-55:

- Referencing to the same section of the guide where the sentence is written:

```diff
- You can still retrieve session cookie and pass to backend like we do [here]
- (#authenticate-kubeflow-pipeline-using-sdk-inside-cluster)
+ You can still retrieve a session cookie and pass to backend.
```

Lines 57-58:

- Refactor:

```diff
- If you are looking for end to end experience, this is working in progress. Once the [feat(sdk): Enable AWS ALB authentication in KFP SDK Client](https://github.com/kubeflow/pipelines/pull/4182) PR is merged,
user can pass Cognito user username and password to authenticate KFP via AWS Application Load Balancer.
+ If you are looking for an end-to-end experience, this is still a work in progress - refer to this PR: [feat(sdk): Enable AWS ALB authentication in KFP SDK Client](https://github.com/kubeflow/pipelines/pull/4182). When it is merged,
you will be able to pass Cognito user username and password to authenticate KFP via AWS Application Load Balancer.
```

Line 85:
- Capitalize "AWS" and use a definite article "the" -> "the AWS secret"

```diff
- If you write any files to S3 in your application, use `use_aws_secret` to attach aws secret to access S3
+ If you write any files to S3 in your application, use `use_aws_secret` to attach the AWS secret to access S3.
```

Line 110:

- Fix typo ("beflow") -> change wording

```diff
- Kubeflow Pipelines supports different artifact viewers. You can create files in S3 and reference them in output artifacts in your application like beflow.
+ Kubeflow Pipelines supports different artifact viewers. You can create files in S3 and reference them in output artifacts in your application as follows:
```

Line 184:

```diff
-  [TensorBoard](/docs/pipelines/sdk/output-viewer/#tensorboard) needs some extra settings on AWS like below:
+  These are the steps you need to take to enable [TensorBoard](/docs/pipelines/sdk/output-viewer/#tensorboard) on AWS:
```

Line 186:

- Refactor:

```diff
- 1. Create a Kubernetes secret `aws-secret` in the `kubeflow` namespace. Follow instructions [here](#s3-access-from-kubeflow-pipelines).
+ 1. Create a Kubernetes secret `aws-secret` in the `kubeflow` namespace by following the instructions in the [S3 Access from Kubeflow Pipelines](#s3-access-from-kubeflow-pipelines) section.
```

Line 188:

- Reformat and refactor:

```diff
- 1. Create a ConfigMap to store the configuration of TensorBoard on your cluster. Replace `<your_region>` with your S3 region.
+ 1. Create a `ConfigMap` to store the TensorBoard configuration on your cluster. Replace `<your_region>` with your S3 region.
```

           name: ml-pipeline-ui
   ```

> 1. By default, NodeInstanceRole has S3 permissions. If users are using customized worker node AMI, ensure that the role attached to the nodegroup running the ml-pipeline-ui pod has read access policy to s3.

Reviewer: A few suggestions:

- Add an extra blank line before the paragraph
- Address the user with "you", rather than referring to them as "users" -> speaking directly to them can help
- Add an indefinite article "a" in "using a customized worker node AMI" and "a read access policy"
- Usually, "make sure"  > "ensure", because it's more formal
- "Node group" > "nodegroup" (unless it's `nodegroup`) - reference: AWS EKS docs https://docs.aws.amazon.com/eks/latest/userguide/managed-node-groups.html
- For variables, values and attribute names from the command line input/output and code, wrap text with backticks (`) "ml-pipeline-ui pod" -> "...`ml-pipeline-ui` pod"
- Capitalize "S" in "S3" (AWS)

```suggestion

1. By default, NodeInstanceRole has S3 permissions. If you are using a customized worker node AMI, make sure that the role attached to the node group running the `ml-pipeline-ui` pod has a read access policy to S3.
```

(or "read-access policy")
